pkg/statistics/region_collection.go

```diff
@@ -240,10 +250,10 @@ func (r *RegionStatistics) Observe(region *core.RegionInfo, stores []*core.Store
 		}
 	}
 	// Remove the info if any of the conditions are not met any more.
-	if oldIndex, ok := r.index[regionID]; ok {
-		deleteIndex = oldIndex &^ peerTypeIndex
+	if oldIndex, ok := r.index[regionID]; ok && oldIndex > 0 {
```

Comment:
for `oldIndex > 0`

0 for region state feels a bit weird, do you think we need to add a state name to 0? Like `notRecord` or something.
and maybe better `!= 0`?

Reply:
For bit operations. `!=` doesn't seem to be used very often